File: cotbeacon/src/main/java/com/jon/cotbeacon/BeaconApplication.java
package com.jon.cotbeacon;

import com.jon.common.AppSpecific;
import com.jon.common.CotApplication;

public class BeaconApplication extends CotApplication {
    @Override
    public void onCreate() {
        super.onCreate();

        /* To supply app-specific info during runtime like package name, build time/version, etc. */
        AppSpecific.setReferenceRepo(new BeaconRepo());
    }
}

File: cotbeacon/src/main/java/com/jon/cotbeacon/BeaconApplication.kt
package com.jon.cotbeacon

import com.jon.common.CotApplication
import com.jon.common.variants.Variant

class BeaconApplication : CotApplication() {
    override fun onCreate() {
        super.onCreate()

        /* To supply app-specific info during runtime like package name, build time/version, etc. */
        Variant.setAppVariantRepository(BeaconRepo())
    }
}
